src/thread.rs

```diff
@@ -803,12 +803,14 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     }
 
     #[inline]
-    fn set_thread_name_wide(&mut self, thread: ThreadId, new_thread_name: Vec<u16>) {
+    fn set_thread_name_wide(&mut self, thread: ThreadId, new_thread_name: &[u16]) {
         let this = self.eval_context_mut();
-        this.machine.threads.set_thread_name(
-            thread,
-            new_thread_name.into_iter().flat_map(u16::to_ne_bytes).collect(),
-        );
+
+        // `GetThreadDescription` isn't implemented, so being lossy is okay.
```

Comment:
GetThreadDescription is a Windows thing, right? This is a core Miri file, not in the shim folder, so please give a bit more context. Imagine someone reading just this file but not knowing about the Windows API at all.